tasks_api/app/controllers/tasks_controller.rb

```diff
@@ -34,16 +34,23 @@ def create
     render json: e, status: :unprocessable_entity
   end
 
+  # rubocop:disable Metrics/AbcSize
   def update
     @task = Task.find(params[:id])
     flash.now['notice'] = I18n.t('notice.updated')
 
-    if @task.update(task_params)
-      render :show
-    else
-      render json: @task.errors, status: :unprocessable_entity
+    Label.transaction do
```

Comment:
この場合、Tasksで例外が起きてもrollbackされるんでしたっけ？:eyes:

Reply:
Task modelのbefore_saveでraiseしてみましたがrollbackされていました。問題なさそうです。
仕様上も大丈夫そうです。

> Though the transaction class method is called on some Active Record class, the objects within the transaction block need not all be instances of that class. This is because transactions are per-database connection, not per-model.


https://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html